src/Native/System.Net.Security.Native/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+project(System.Net.Security.Native)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+add_definitions(-DPIC=1)
+
+if (HAVE_HEIMNTLM_HEADERS)
+   find_library(LIB_HEIMNTLM NAMES heimntlm PATHS /usr/heimdal/lib)
+   include_directories(/usr/heimdal/include)
+
+   if(LIB_HEIMNTLM STREQUAL LIB_HEIMNTLM-NOTFOUND)
+       message(FATAL_ERROR "Cannot find libheimntlm.so and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+   endif()
+
+
+   set(NATIVENTLM_SOURCES
+     pal_ntlmapi.cpp
+   )
+
+   add_library(System.Net.Security.Native
+    SHARED
+    ${NATIVENTLM_SOURCES}
+   )
+
+   target_link_libraries(System.Net.Security.Native
+           ${LIB_HEIMNTLM}
+           ${OPENSSL_CRYPTO_LIBRARY}
+   )
+
+   install (TARGETS System.Net.Security.Native DESTINATION .)
+
+elseif (CMAKE_SYSTEM_NAME STREQUAL Linux)
+   message (FATAL_ERROR "cannot find libheim ntlm headers and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+else ()
```

Comment:
@stephentoub I'm... having trouble parsing that.

* Interpretation 1: The Ubuntu/CentOS package(s) related to NTLM support have functioning libraries, but the dev packages conflict in unusual ways with the dev packages for other libraries we use (such as libcurl)
* Interpretation 2: The Unbutu/CentOS package(s) related to NTLM are completely in conflict with libcurl, and you have to choose one or the other

Interpretation 1 seems more likely given the context (though, more confusing).  But I'm also not sure what state you want to end up in... I think it's "let Linux machines which are missing the headers just not build this library, like when coreclr had the capability of 'succeeding' without building System.Security.Cryptography.Native".  In that case, how do we accomplish:

1) Ensuring that the official builds successfully produce this library
2) Assuming that this library is present, and that we've written tests, run the tests
3) Assuming that when I (or other non-official-build-machine-entity) build there's no/a-broken library produced, not emit errors when doing BuildAndTest locally

Everything that I can think of to solve (3) impacts the ability to do (1).

If "short term" means "99.97% likely to be solved before RTM", then I can understand your proposal (1).  Otherwise something like (2) (I'd add NetSecurity_IsNtlmSupported, which turns into a PlatformNotSupportedException in the appropriate place(s) when it returns false).